### PR TITLE
[stable28] chore: update workflows from templates

### PR DESCRIPTION
**Pull request by nextcloud-command**
This PR was automatically generated by the workflow dispatcher
Please check that there are no reverted changes before merging ⚠

See templates in https://github.com/nextcloud/.github

### CODE REVIEW

File: .github/workflows/phpunit-mysql.yml [hunk 1]
```diff
@@ -33,17 +15,47 @@ concurrency:
...
     runs-on: ubuntu-latest
 
+    needs: changes
+    if: needs.changes.outputs.src != 'false'
+
     strategy:
       matrix:
-        php-versions: ['8.0', '8.1', '8.2']
+        php-versions: ['8.0', '8.1', '8.2', '8.3']
         server-versions: ['master']
```
Comment (come-nc) [inline .github/workflows/phpunit-mysql.yml:51]:
```suggestion
        server-versions: ['stable28']
```

File: .github/workflows/phpunit-pgsql.yml [hunk 2]
```diff
@@ -33,12 +15,39 @@ concurrency:
...
     runs-on: ubuntu-latest
 
+    needs: changes
+    if: needs.changes.outputs.src != 'false'
+
     strategy:
       matrix:
-        php-versions: ['8.0']
+        php-versions: ['8.2']
         server-versions: ['master']
```
Comment (come-nc) [inline .github/workflows/phpunit-pgsql.yml:51]:
```suggestion
        server-versions: ['stable28']
```

File: .github/workflows/phpunit-sqlite.yml [hunk 3]
```diff
@@ -33,12 +15,39 @@ concurrency:
...
     runs-on: ubuntu-latest
 
+    needs: changes
+    if: needs.changes.outputs.src != 'false'
+
     strategy:
       matrix:
-        php-versions: ['8.0']
+        php-versions: ['8.2']
         server-versions: ['master']
```
Comment (come-nc) [inline .github/workflows/phpunit-sqlite.yml:51]:
```suggestion
        server-versions: ['stable28']
```

**Review by come-nc — CHANGES_REQUESTED** [review]

**Comment by provokateurin** [comment]
Covered in https://github.com/nextcloud/external/pull/587

**Review by come-nc — APPROVED** [review]

**Pull request merged**